### PR TITLE
Uplink Additions: The Hitman DLC

### PR DESCRIPTION
**Pull request by TheDanseMacabre**
Ever notice that your options are limited when it comes to stealthy guns? Not anymore! This PR adds two new guns to the uplink. The first is an integrally suppressed machine pistol - this gun comes in a new type of ammo, 6mm, which is low damage and completely ineffective against armor, but the magazine capacity is large and the rapid rate of fire means that against unarmored and unsuspecting targets, ambushes can be very very successful. The other is a gun that looks like a camera at a glance until you deploy it, at which point it becomes an ordinary 9mm pistol.

SMG sprite is from Eris and has been edited to fit in more with Aurora's palette. The camera gun uses terrible codersprites, but it'll do until I can find a replacement.

### CODE REVIEW

**Comment by TheDanseMacabre** [comment]
!review

File: code/modules/projectiles/guns/projectile/automatic.dm [hunk 1]
```diff
@@ -617,6 +617,26 @@
...
+	caliber = "6mm"
+	suppressed = TRUE
+	can_unsuppress = FALSE
+	handle_casings = DELETE_CASINGS
+	max_shells = 30
+	allowed_magazines = list(/obj/item/ammo_magazine/submachinemag/assassin)
+
+/obj/item/gun/projectile/automatic/tommygun/assassin/update_icon()
+	..()
+	icon_state = (ammo_magazine)? "assassin_smg" : "assassin_smg-empty"
```
Comment (FluffyGhoster) [inline code/modules/projectiles/guns/projectile/automatic.dm:638]:
```suggestion
	icon_state = (ammo_magazine) ? "assassin_smg" : "assassin_smg-empty"
```

File: code/modules/projectiles/guns/projectile/shotgun.dm [hunk 2]
```diff
@@ -255,3 +255,21 @@
 		toggle_folded(user)
 		return FALSE
 	return ..()
+
+/obj/item/gun/projectile/shotgun/foldable/cameragun
+	name = "camera"
+	desc = "A polaroid camera"
+	desc_extended = "Upon closer inspection, this is not a camera at all, but a 9mm firearm concealed inside the shell of one, which can be deployed by pressing a button."
```
Comment (FluffyGhoster) [inline code/modules/projectiles/guns/projectile/shotgun.dm:262]: This should probably be hidden and only shown if the camera is inspected in hand or the likes, this way anyone from afar can shift click you, select the camera if you have it in hand, then check the extended description and know it's a firearm

Comment (TheDanseMacabre) [reply code/modules/projectiles/guns/projectile/shotgun.dm:262]: If there's a way to do that I'd be happy to make that happen

Comment (FluffyGhoster) [reply code/modules/projectiles/guns/projectile/shotgun.dm:262]: > If there's a way to do that I'd be happy to make that happen

/atom/proc/examine_fluff(mob/user, distance, is_adjacent, infix = "", suffix = "")

You can override this proc to override what extended description is displayed

Comment (Arrow768) [inline code/modules/projectiles/guns/projectile/shotgun.dm:262]: As fluffy mentioned above, that's going to have to be changed.

**Review by Arrow768 — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/projectile/shotgun.dm (hunk 2 above)
Comment (TheDanseMacabre) [reply code/modules/projectiles/guns/projectile/shotgun.dm]: Done

**Review by NonQueueingMatt — APPROVED** [review]

**Review by DreamySkrell — APPROVED** [review]

**Review by FluffyGhoster — APPROVED** [review]

**Pull request merged**